api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
So given the following exampe:

```go
type DebuggerContext struct {
    // Requested is the desired status of the debugger.
    Requested DebuggerStatus `json:"requested,omitempty"`
    // Status is the actual status of the debugger.
    Status DebuggerStatus `json:"status,omitempty"`
}

// DebuggerStatus represents debugger status.
type DebuggerStatus struct {
    // Running is true if the debugged process is continuing execution, and is
    // false if the debugged process is currently suspended.
    Running bool `json:"running"`
    // The action we are trying to perform, i.e. 'stepping', 'continuing', 'nexting', 'stopping'.
    Action string `json:"action"`
    // BreakPoint is the current breakpoint at which the debugged process is
    // suspended, and may be empty if the process is not suspended. This is
    // a read-only field.
    BreakPoint BreakPoint `json:"breakPoint,omitempty"`
    // Thread is the currently selected debugger thread.
    Thread Thread `json:"thread"`
}
```

The only piece of state we expect to diverge here is `Running`. As far as the state of the process goes, we only request that it start, i.e. `step`, `continue`, `next` and stop (for ctrl-c). I'm still failing to see the need for a full `Requested` field, since once we make that request, the actual `Status` field should have moved to `Status.Running = true` and `Status.Action = "stepping" // for example`.

I'm failing to see when the requested state would be useful to hold onto for either the client or the server since only one (or two) entries in the struct will be modified (`running`, `action`).